source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+		/**
+		* @private
+		*/
+		valueChanged: function () {
+			if (this.value == null || this.value === '') {
+				this.resetPicker();
+				this.noneTextChanged();
+				this.doDurationChange({name: this.name, value: this.$.currentValue.content});
+			} else {
+				this.createValueArray();
+				this.updatePicker();
+				this.$.currentValue.set('content', this.formatText());
+			}
+		},
+
+		/**
+		* @private
+		*/
+		createValueArray: function () {
+			var tempValue;
+			if (this.value != null && this.value !== '') {
+				var timeArray = this.value.toString().split(':');
+				for (var idx = 0; idx < this.templateArray.length; idx++) {
+					var item = this.templateArray[idx];
+					switch (item) {
+						case 'h':
+						if (this.$.hourPicker || !this.$.hourPicker && timeArray.length > 2) {
+							tempValue = Math.round(timeArray.shift());
+							if (isNaN(tempValue) || tempValue > 23 || tempValue < 0) {
+								tempValue = 0;
+							}
+							this.valueArray['hour'] = tempValue;
+						}
+						break;
+						case 'm':
+						if (this.$.minutePicker || !this.$.minutePicker && timeArray.length > 1) {
+							tempValue = Math.round(timeArray.shift());
+							if (isNaN(tempValue) || tempValue > 59 || tempValue < 0) {
+								tempValue = 0;
+							}
+							this.valueArray['minute'] = tempValue;
+						}
+						break;
+						case 's':
+						if (this.$.secondPicker || !this.$.secondPicker && timeArray.length > 0) {
+							tempValue = Math.round(timeArray.shift());
+							if (isNaN(tempValue) || tempValue> 59 || tempValue < 0) {
+								tempValue = 0;
+							}
+							this.valueArray['second'] = tempValue;
+						break;
+						}
+					}
+				}
+			}
+		},
+
+		/**
+		* @private
+		*/
+		updatePicker: function () {
+			if (this.$.hourPicker && this.$.hour.value != this.valueArray['hour'])
+				this.$.hour.set('value', this.valueArray['hour']);
+
+			if (this.$.minutePicker && this.$.minute.value != this.valueArray['minute'])
+				this.$.minute.set('value', this.valueArray['minute']);
+
+			if (this.$.secondPicker && this.$.second.value != this.valueArray['second'])
+				this.$.second.set('value',this.valueArray['second']);
+		},
+
+		/**
+		* @private
+		*/
+		formatValue: function () {
+			var item, idx, val = '';
+			for (idx = 0; idx < this.templateArray.length; idx++) {
+				item = this.templateArray[idx];
+				switch (item) {
+					case 'h':
+						val = val + ':' + this.valueArray['hour'];
+						break;
+					case 'm':
+						val = val + ':' + this.valueArray['minute'];
+						break;
+					case 's':
+						val = val + ':' + this.valueArray['second'];
+						break;
+				}
+			}
+			return val.substring(1);
+		},
+
+		/**
+		* @private
+		*/
+		formatText: function () {
+			var item, idx, text = '';
+			if (this.valueArray) {
+				for (idx = 0; idx < this.templateArray.length; idx++) {
+					item = this.templateArray[idx];
+					switch (item) {
+						case 'h':
+							text = text + ' ' + this.valueArray['hour'] + ' ' + moon.$L('Hours');
```

Comment:
We need to take into account number agreement between the value and the units (i.e. 1 Hour).